src/yarp.c

```diff
@@ -2154,6 +2157,49 @@ parse_arguments_list(yp_parser_t *parser, yp_arguments_t *arguments) {
       break;
     }
   }
+
+  if (accept(parser, YP_TOKEN_BRACE_LEFT) || accept(parser, YP_TOKEN_KEYWORD_DO)) {
+    not_provided(parser);
+    not_provided(parser);
+
+    yp_token_t opening = parser->previous;
+
+    accept(parser, YP_TOKEN_NEWLINE);
+
+    yp_node_t *block_arguments = yp_node_arguments_node_create(parser);
```

Comment:
I think we want to parse parameters here, not arguments. Because you can do things like

```ruby
foo do |bar:, baz:|
end
```